database/rrdhost.c

```diff
@@ -401,6 +401,33 @@ RRDHOST *rrdhost_find_or_create(
                 , system_info
                 , 0
         );
+
```

Comment:
This procedure is only called from one place (ignoring the unit-test entry points) - inside `rrdpush_receive()`, which is only called from `rrdpush_receiver_thread()`.

The way it is implemented in this PR, receiving a STREAM will cause the flag to flip to true. But if the stream is closed then the flag will not flip back to false. There are two ways for how to implement this:
1. Statically decide if we are a master based on the loaded configuration (then we don't need this code).
2. Dynamically decide if we are master because there is an incoming stream (then we need something more complex than this).

Option (2) is complicated in the presence of multiple incoming streams. @cakrit how do you want this implemented: fully dynamic view of whether we are receiving metrics, or a static view of if we are expecting metrics?

Reply:
We only discussed Option 1 for both slave and master. Making it dynamic may keep flipping the label, which is not very useful.

Reply:
I didn't find an easy way to get information from the configuration directly. That's why I put the code into `rrdhost_find_or_create()`.

Reply:
Then the solution is simplest - we delete this piece of code and put something in the startup sequence that determines from the config whether the node is a master or not.

Reply:
What if the config is wrong and Netdata never receives data from slaves? Will we treat the node as a master?

Reply:
It seems it is `is_configured_as_master` label

Reply:
Yes - if the configuration is wrong and the node never receives data from slaves then we will still consider it to be a master. 

This is significantly easier than the alternative, which (as @cakrit points out above) will lead to a poor user experience as labels will flip states and so they will not be able to use them to filter nodes, which is the desired purpose of this feature.

To do the alternative properly we would need a counter of how many slaves are connected, rather than just a simple flag and it would be be a much more complex implementation for something that would look worse when the user tries to access the functionality by filtering the nodes.

Think of it as `is_configured_as_master` as @ilyam8 put it - but keep the specified name `_is_master`.

Reply:
It has been rewritten. Now the configuration file is used to check if a node was configured as a master.